src/endpoint/blob/blob_utils.js

```diff
@@ -53,6 +55,60 @@ function parse_copy_source(req) {
 }
 
 
+
+
+//////// BLOB API UTILS ////////
+
+// https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs
+// This API called directly because of an inefficient matching function of the Azure blob SDK.
+async function list_objects(params, account_name, container, sasToken) {
+
+    const hostname = `${account_name}.blob.core.windows.net`;
+    const delimiter = params.delimiter || '/';
+    const maxResults = params.limit === 0 ? 1 : params.limit;
+
+    let path = `https://${account_name}.blob.core.windows.net/${container}?restype=container&comp=list&delimiter=${delimiter}` +
+        `&maxresults=${maxResults}&${sasToken}`;
+    if (params.key_marker) path += `&marker=${params.key_marker}`;
+
+    let response;
+    try {
+        response = await make_http_request({ hostname, port: 443, path, method: 'GET' }, undefined, undefined);
+    } catch (err) {
+        throw new Error(`GET ${path} did not responed or returned with an error ${err}`);
+    }
+    const status_code = response.statusCode;
+    const buffer = await read_stream_join(response);
+    const body = buffer.toString('utf8');
+    if (status_code !== 200) {
+        throw new Error(`Could not get blobs and diresctories list, (status code: ${status_code}) got ${body}`);
+    }
+
+    let blobs;
+    let dirs;
+    let next_marker;
+    try {
+        const parsed = await parse_xml_to_js(body, undefined);
```

Comment:
that undefined `options` argument is optional anyway and it is less cluttered to read without it.

Reply:
deleted